### ISSUE TITLE
Compiled app hangs on centos 6.8

### ISSUE THREAD

**Issue by jsjie** [JuliaLang/PackageCompiler.jl#720]
using docker image [centos 6.8](https://hub.docker.com/layers/library/centos/6.8/images/sha256-82f3b5f3c58f22e50d6b05f227c675af504cffc9dff7e318df5fc40faee6410e?context=explore)
and julia 1.7.1 official binary, a hello-world package with only one file in src/:
```
module HW
function hw()::Cint
    print("hello, world!")
    return 0
end
end
```

The compiled binary will hang in the image. However, copy it to something like ubuntu 20.04, and it works just fine.

**Comment by p20241213** [comment]
@jsjie did you ever figure out the issue? mine hangs in ubuntu and debian but works in windows  :(

**Comment by jsjie** [comment]
No, I didn't fix it. It has been too long and I cannot remember clearly, but it may be due to glibc. Julia has a requirement for the glibc version, and that on Centos 6.8 is too old. Currently my app runs fine on Centos 7.
